lib/BuildSystem/BuildSystem.cpp

```diff
@@ -3309,6 +3338,10 @@ class StaleFileRemovalCommand : public Command {
     llvm::raw_svector_ostream(result) << "]";
   }
 
+  virtual basic::CommandSignature getSignature() const override {
+    return {};
```

Comment:
Is this safe, shouldn't it have some unique type code assigned to it (so rule type changes can be detected)? Actually, the same comment applies to all the other places a null signature is created, maybe they should each have a unique code.

Reply:
For many of the discrete key types it is unnecessary as the key itself would change. I think the same applies to these commands as well, but I'll take a second look.

Reply:
Yeah, after closer inspection I agree these should have signatures. I've added a default implementation for the Command object and removed these null signature overrides.